software/contrib/consequencer.md

```diff
@@ -97,16 +106,23 @@ A fixed voltage will select a single pattern and varying voltage (e.g. an envelo
 
 Patterns can be added, removed or updated by updating the relevant list structures at the end of the consequencer.py file in the pattern class (look for the line `class pattern:`). Once patterns are updated make sure you save a copy of the updated file to the EuroPi using yoru favourite method (Thonny / REPL), then restart the Consequencer script.
 
-The syntax should be intuitive. A four-on-the-floor exanmple pattern is shown below. Each `1` or `0` represents a gate or no gate at that point in the sequence.
+The syntax should be intuitive. An exanmple pattern is shown below. Each `1` or `0` represents a gate or no gate at that point in the sequence.
 The mapping of `BD`, `SN`, `HH` is as follows:
 - BD: Output 1
 - SN: Output 2
 - HH: Output 3
 
+Starting with the July 2022 update, steps now also have a probability which is configured using the BdProb, SnProb and HhProb sections as shown beloe.
+Use a value from `1` to `9` to set the desired level of probability for each step.
+A value of `9` will cause the step to trigger 100% of the time, any value from `1` to `8` will trigger the step from n/9 times.
```

Comment:
[discussion] the 1 through 9 scale with 9 being 100% and 11% buckets feels unintuitive to me. However, the most obvious alternative, while still using a single int, is to use 1 through '10', with 0 (as 10) representing 100%. This scale has the advantage of more common 10% buckets, and the ability to set a step at 50%, but the disadvantage of 0 meaning 100%.

Reply:
The decision to use 1-9 is partly lazyness, but mainly to make it easier to enter probability values in patterns.
Having just one digit allows consequencer to index each char without needing to explicitly create an array.
i.e. pattern['949'] can be indexed as pattern[0], pattern[1], etc. If I used values up to 10, pattern would need to be written as pattern[9, 4, 9] which is slightly more cumbersome.

Reply:
I was suggesting that the digit 0 could represent 10. Which in itself would be weird. Up to you to decide between that weirdness or the 9 step range.

Reply:
Apologies, I should have read your comment more slowly!
I too think it is a little unintuitive having values from 1-9. However, I also like the way it forces an irregular probability -  25% / 50% etc just feels too normal for Eurorack :)